### PR TITLE
Hb migration - web part

### PR DESCRIPTION
**Pull request by codci**
* update WebDriver part modules
* add new web module test sample
* small fixes

### CODE REVIEW

File: common/_libs/helpers/singleton.py [hunk 1]
```diff
@@ -23,23 +23,44 @@ def get_singleton_class(key_name, base_class):
...
+
+
+def delete_singleton_object(base_class):
     """Remove class instance with specified name and inheritable base class
        from Singleton instances.
 
     Args:
-        key_name (str): Class name
-        base_class (str): Base class name
+        base_class (type): Base class name
```
Comment (maximrodin) [inline common/_libs/helpers/singleton.py:60]: Remove name

File: common/_webdriver_qa_api/core/base_pages.py [hunk 2]
```diff
@@ -16,35 +17,22 @@ def __init__(self, driver, config, locator_type=None, locator=None,
...
         assert page present, test fails if page is absent
         """
-        if not silent:
-            logger.info(f"Verify is page '{self.name}' opens in {second} seconds")
-        page_element = BaseElement(self.locator_type, self.locator, self.driver, self._config)
-        assert_should_be_equal(actual_value=page_element.is_present_without_waiting,
-                               expected_value=True, silent=silent, timeout=second)
-        if not silent:
-            logger.info(f"Page '{self.name}' is opened")
+        assert_should_be_equal(actual_value=self.is_page_present(second=second), expected_value=True,
```
Comment (maximrodin) [inline common/_webdriver_qa_api/core/base_pages.py:24]: Perfect!

File: common/_webdriver_qa_api/core/dropdown_mixin.py [hunk 3]
```diff
@@ -1,25 +1,26 @@
...
 
     @property
-    def menu_items(self):
+    def menu_items(self) -> List['WebElement']:
         return self.menu.element().find_elements(By.CLASS_NAME, 'dropdown-item')
 
     def open_menu(self):
         self.click()
-        self.menu.wait_element_visible()
+        self.menu.assert_visible()
```
Comment (maximrodin) [inline common/_webdriver_qa_api/core/dropdown_mixin.py]: We expect that menu will be visible without any timeout?

File: common/_webdriver_qa_api/core/text_box_mixin.py [hunk 4]
```diff
@@ -21,51 +23,39 @@ def set_text(self, text, skip_if_none=True, blur_and_focus=False):
...
+    def assert_is_text_masked(self, is_masked: bool = True):
         """
         assert element text masked, test fails if expected state isn't equal to actual
         :param is_masked: if true - should be masked, if false - not masked
         """
-        if is_masked:
-            expected_state = "true"
-        else:
-            expected_state = "false"
+        # expected_state = "true" if is_masked else "false"
```
Comment (maximrodin) [inline common/_webdriver_qa_api/core/text_box_mixin.py]: Delete?

File: common/_webdriver_qa_api/web/web_elements.py [hunk 5]
```diff
@@ -1,52 +1,61 @@
...
+                         driver=self.driver, config=ConfigManager(), name=name,
                          parent=parent)
         self.ALLOWED_DYNAMIC_METHODS = ["click", "text"]
 
-    def assert_element_placeholder(self, expected):
+    def __getattribute__(self, item):
+        if item == 'element':
+            if self.frame is not None:
+                frame_element = self.frame.element.selenium_element
+                self.driver.switch_to.frame(frame_element)
```
Comment (maximrodin) [inline common/_webdriver_qa_api/web/web_elements.py:34]: If I understand correctly frame it's IFrame. Could you please add a docstring please?

**Review by maximrodin — COMMENTED** [review]
You can merge this change. But comments may be fixed later

File: common/_webdriver_qa_api/core/dropdown_mixin.py (hunk 3 above)
Comment (codci) [reply common/_webdriver_qa_api/core/dropdown_mixin.py]: wait_element_visible looks better, revert to this option

File: common/_webdriver_qa_api/web/web_elements.py (hunk 5 above)
Comment (codci) [reply common/_webdriver_qa_api/web/web_elements.py:34]: docstring was added

File: common/_webdriver_qa_api/core/text_box_mixin.py (hunk 4 above)
Comment (codci) [reply common/_webdriver_qa_api/core/text_box_mixin.py]: revert to commented option, also add todo comment for this method.
I'm not sure that it's a universal solution and we need to double check how to work with masked fields

File: common/_libs/helpers/singleton.py (hunk 1 above)
Comment (codci) [reply common/_libs/helpers/singleton.py:60]: done

**Review by maximrodin — APPROVED** [review]
You can merge this PR

**Pull request merged**